controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -54,29 +57,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	cachedCluster, ok := cluster.GetCachedToolchainCluster(toolchainCluster.Name)
 	if !ok {
 		err := fmt.Errorf("cluster %s not found in cache", toolchainCluster.Name)
-		toolchainCluster.Status.Conditions = []toolchainv1alpha1.Condition{clusterOfflineCondition()}
+		toolchainCluster.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(toolchainCluster.Status.Conditions, clusterOfflineCondition())
 		if err := r.Client.Status().Update(ctx, toolchainCluster); err != nil {
 			reqLogger.Error(err, "failed to update the status of ToolchainCluster")
 		}
 		return reconcile.Result{}, err
 	}
 
+	//Not testing as this as this is being tested in cluster package
 	clientSet, err := kubeclientset.NewForConfig(cachedCluster.RestConfig)
 	if err != nil {
 		reqLogger.Error(err, "cannot create ClientSet for the ToolchainCluster")
+		toolchainCluster.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(toolchainCluster.Status.Conditions, clusterNotReadyCondition())
```

Comment:
Updating the conditions in the structure without updating the status won't have any effect. So, we should either persist the change with `r.Client.Status().Update(...)` (maybe `r.updateStatus`?) or we can remove this line.

Reply:
Can you elaborate more..? 
My understanding here is we update the status every time we poll to check healtth,  but inbetween we still want to have the condition updated if something breaks while checking the health. so we are updating the condition and not status.

Reply:
sure, basically L71 is updating the data in memory but we're never reaching out the database to persist this change because L72 returns. To persist the data in the database we need to do something like L60-L64.

It seems to me we have an auxiliary function that implements this logic (`r.updateStatus`) we can simplify and just invoke `r.updateStatus(ctx, toolchainCluster, clusterNotReadyCondition())`.

Reply:
thanks for pointing this out,  i have changed it